services/replicator/replicatorclientfactory.go

```diff
@@ -170,7 +171,12 @@ func (f *repCltFactoryImpl) GetHostsForAllDeployment() map[string][]string {
 
 func (f *repCltFactoryImpl) UpdateHostsForDeployment(deployment string, hosts []string) {
 	f.hostLk.Lock()
+	f.deploymentClientsLk.Lock()
+	defer f.deploymentClientsLk.Unlock()
 	defer f.hostLk.Unlock()
```

Comment:
you probably want to move the 'defer hostLk.Unlock()' to be immediately after the "h.hostLk.Lock()' statement. (the 'defers' are processed LIFO order ..)